packages/devtools_app/lib/src/http/http_request_data.dart

```diff
@@ -69,18 +69,24 @@ class HttpRequestData extends NetworkRequest {
       }
     }
 
+    // Stitch together response as it may have been sent in multiple chunks.
+    final encodedData = <int>[];
     for (final event in responseEvents) {
       final traceEvent = TraceEvent(event);
       // TODO(kenz): do we need to do something with the other response events
       // (phases 'b' and 'e')?
       if (traceEvent.phase == TraceEvent.asyncInstantPhase &&
           traceEvent.name == 'Response body') {
-        final encodedData = (traceEvent.args['data'] as List).cast<int>();
-        responseBody = utf8.decode(encodedData);
-        break;
+        encodedData.addAll((traceEvent.args['data'] as List).cast<int>());
       }
     }
 
+    try {
+      responseBody = utf8.decode(encodedData);
+    } on FormatException {
+      // Non-UTF8 response.
```

Comment:
Should we do something to handle non UTF8 responses so that the user has some indication that we did receive a response? What is an example of a non-utf8 response?

Reply:
An image is an example of a non-UTF8 response. If we try and decode that, it's certain to blow up.